src/Pipe.jl

```diff
@@ -5,45 +5,62 @@ export @pipe
 
 const PLACEHOLDER = :_
 
-function rewrite(ff::Expr,target)
-    function replace(arg::Any)
-        arg #Normally do nothing
+function replace(arg::Any, target)
+    arg #Normally do nothing
+end
+
+function replace(arg::Symbol, target)
+    if arg==PLACEHOLDER
+        target
+    else
+        arg
     end
-    function replace(arg::Symbol)
-        if arg==PLACEHOLDER
-            target
-        else
-            arg
+end
+
+function replace(arg::Expr, target)
+    rep = copy(arg)
+    rep.args = map(x->replace(x, target), rep.args)
+    rep
+end
+
+function rewrite(ff::Expr, target, broadcast=false)
+    if broadcast
+        temp_var = gensym()
+        rep_args = map(x->replace(x, temp_var), ff.args)
+        if ff.args != rep_args
+            #_ subsitution
+            ff.args = rep_args
+            return :($temp_var->$ff)
+        end
+    else
+        rep_args = map(x->replace(x, target), ff.args)
+        if ff.args != rep_args
+            #_ subsitution
+            ff.args = rep_args
+            return ff
         end
-    end
-    function replace(arg::Expr)
-        rep = copy(arg)
-        rep.args = map(replace,rep.args)
-        rep
     end
 
-    rep_args = map(replace,ff.args)
-    if ff.args != rep_args
-        #_ subsitution
-        ff.args=rep_args
-        return ff
-    end
     #No subsitution was done (no _ found)
     #Apply to a function that is being returned by ff,
     #(ff could be a function call or something more complex)
-    rewrite_apply(ff,target)
+    rewrite_apply(ff,target,broadcast)
 end
 
-
-function rewrite_apply(ff, target)
-    :($ff($target)) #function application
+function rewrite_apply(ff, target, broadcast=false)
```

Comment:
can we break this into two functions?
`rewrite_apply` and `rewrite_apply_broadcasted`